scheduler.py

```diff
@@ -146,8 +193,43 @@ def _run():
                     )           
                     run_logger.record_run(identifier, run_datetime)
                 except Exception:
-                	traceback.print_exc()
-                	continue
+                    print "Error in generating report %s" % identifier
+                    error_list.add_error("Error in generating report %s : \n %s" % (identifier, traceback.format_exc()))
+                    traceback.print_exc()
+                    continue
+
+def dry_run():
```

Comment:
There's a fair bit of repeated code between this and the `_run` function.  Would it be possible to unify the two with an optional `dry_run` keyword argument?  That way, we know that dry run is definitely a dry run of the exact logic that will be used in the real run